httpserver/http_connection.cpp

```diff
@@ -87,6 +87,78 @@ void make_http_request(boost::asio::io_context& ioc, std::string sn_address,
     make_http_request(ioc, sn_address, port, req, cb);
 }
 
+static void parse_swarm_update(const std::shared_ptr<std::string>& response_body, const swarm_callback_t& cb) {
+    const json body = json::parse(*response_body, nullptr, false);
+    if (body == nlohmann::detail::value_t::discarded) {
+        BOOST_LOG_TRIVIAL(error) << "Bad lokid rpc response: invalid json";
+        return;
+    }
+    all_swarms_t all_swarms;
+    std::map<swarm_id_t, std::vector<sn_record_t>> swarm_map;
+
+    try {
+        const json service_node_states_string = body["result"]["as_json"];
+        const std::string list_string = service_node_states_string.get<std::string>();
+        const json service_node_states = json::parse(list_string, nullptr, false);
+
+        auto service_node_it = service_node_states.begin();
+        while(service_node_it != service_node_states.end()) {
```

Comment:
How about `for (const auto &sn : service_node_states) {...}`

Reply:
Ah thank you, I didn't like the while loop but didn't realise I could do this syntax